AsyncDisplayKit/Details/ASCollectionDataController.h

```diff
@@ -30,6 +30,10 @@
 
 - (NSUInteger)dataController:(ASCollectionDataController *)dataController supplementaryNodesOfKind:(NSString *)kind inSection:(NSUInteger)section;
 
+@optional
+
+- (ASDataControllerCellNodeBlock)dataController:(ASCollectionDataController *)dataController supplementaryNodeBlockOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath;
```

Comment:
Wow, you really went all the way here! I had implicitly assumed we would leave supplementary node creation synchronous, but this is great to have. Definitely a motivator to do that refactor to combine in the whale code for supplementary and regular cell nodes, while also adding support for headers and footers… upshot is, if blocks for supplementary nodes in particular are adding complexity, feel free to nuke them from this current effort.